Windows Thrdscan: Fix broken tuple unpacking

dgmcdona opened the pull request:
Timeliner fails due to an incorrect unpacking of this tuple, which needs
3 additional dictionary items for start path, win32 start path, and
win32 start address.

dgmcdona committed:
Windows Thrdscan: Fix broken tuple unpacking

Timeliner fails due to an incorrect unpacking of this tuple, which needs
3 additional dictionary items for start path, win32 start path, and
win32 start address.

ikelos commented:
Could we please get a minimal test with the existing linux image as part of the PR to ensure this doesn't happen again please?

dgmcdona committed:
Windows Timeliner: Add basic test

This is enough to ensure that the return code is nonzero and there was
some valid output.

dgmcdona commented:
I added a test locally, but it took ~226 seconds to run. I'm happy to push that if you're okay with adding that much time to test runs, but what if we also/instead updated `_generator()` to return a named tuple instead, and access the values by field name in `generate_timeline()`? That would make this kind of bug a little harder to trigger.

ikelos commented:
Errrr, yeah, I'm happy with that?  I dunno if that should become a core coding style (in which case we should add it to the `CODING_STYLE.md` and explain the rationale so people know *why* they're doing it).  It might also bulk out a bunch of plugins unnecessarily, so we might want to figure out the cheapest way of doing it (namedtuples/dataclasses/etc).

atcuno commented:
Tests taking another 4 minutes seem fine with me to avoid these issues.

ikelos reviewed (approved):
Ok, looks good, thanks  5:)

ikelos commented:
We can still do the namedtuple change later if you think it'd be an improvement, but I'd like to keep API churn to a minimum so only if you think it'd be valuable.
